### PR TITLE
Interfaces for configuration/event objects and build fixes

### PR DESCRIPTION
**Pull request by morabitowoolpert**

### CODE REVIEW

**Comment by BTMorton** [comment]
You have a very aggressive text editor when it comes to formatting!

File: src/NgGrid.ts [hunk 1]
```diff
@@ -64,29 +85,29 @@ export class NgGrid implements OnInit, DoCheck, OnDestroy {
...
-		'visible_rows': 0,
-		'col_width': 250,
-		'row_height': 250,
-		'cascade': 'up',
-		'min_width': 100,
-		'min_height': 100,
-		'fix_to_grid': false,
-		'auto_style': true,
-		'auto_resize': false
+	private static CONST_DEFAULT_CONFIG = <NgGridConfig>{
```
Comment (BTMorton) [inline src/NgGrid.ts]: Is there a reason for this over `private static CONST_DEFAULT_CONFIG: NgGridConfig = {` ?

Comment (morabitowoolpert) [inline src/NgGrid.ts]: No, you're right, it should be CONST_DEFAULT_CONFIG: NgGridConfig

**Comment by BTMorton** [comment]
Thanks very much! :)

**Pull request merged**